dataservice/extensions/flask_indexd.py

```
@@ -55,9 +55,14 @@ def get(self, record):
             return record
 
         url = self.url + record.latest_did
-        resp = self.session.get(url)
-        self.check_response(resp)
-        resp.raise_for_status()
+        try:
+            resp = self.session.get(url)
+            self.check_response(resp)
+            resp.raise_for_status()
+        except RecordNotFound as err:
+            raise err
```

Why raise an exception here instead of calling `abort`? 

And if you really do want to raise the exception then you could prob just call `check_response` outside the try except

---

There's some round-about logic that relies on the specific `RecordNotFound` exception when trying to resolve documents that no longer exist:
https://github.com/kids-first/kf-api-dataservice/blob/b81e68648ec9d1a880deb1e6970862a6e46199e5/dataservice/api/common/model.py#L136-L140

---

I could move the `check_response()` call outside of the try/except block, but then I will need two try/excepts for both the check_response()` and `raise_for_status()`.

---

> There's some round-about logic that relies on the specific RecordNotFound exception when trying to resolve documents that no longer exist

Ohhh ok, I forgot about that 👍

---

> I could move the check_response() call outside of the try/except block, but then I will need two try/excepts for both the check_response() and raise_for_status().

Why? `check_response` just raises RecordNotFound which you're just re-raising when you catch it right?

---

yeah, you're right

---

Actually, I would still need to try/catch blocks because both `raise_for_status()` and `get()` may raise `RequestException`.

---

I will just get rid of the `RecordNotFound` handler here.